src/main/java/uk/nhs/ctp/audit/AuditService.java

```diff
@@ -50,7 +50,9 @@ public void startEntry(HttpRequest request) {
 
     auditThreadStore.getCurrentAuditSession()
         .orElseThrow(IllegalStateException::new)
-        .getEntries().add(entry);
+        .toBuilder()
```

Comment:
This looks like the source of the missing entries - converting to a builder rather than mutating the object in place. The result needs assigning back to the current audit session.

Reply:
was just about to comment this!

Reply:
:) my bad, had some issues with lombok and singular builder as `@Builder` just assumes the class it annotates will be immutable; I've forgotten to add it back in, fixed now - thank you!